cli/tests/pcluster/createami/test_pcluster_createami.py

```diff
@@ -0,0 +1,118 @@
+"""This module provides unit tests for (portions of) the `pcluster createami` code."""
+
+import os
+
+import pytest
+
+import pcluster.createami as createami
+from assertpy import assert_that
+from pcluster.constants import SUPPORTED_ARCHITECTURES
+from recordclass import recordclass
+
+MockedCreateAmiArgs = recordclass(
+    "MockedCreateAmiArgs", ["base_ami_id", "instance_type", "base_ami_os"], rename=False, defaults=None, module=None
+)
+
+
+@pytest.mark.parametrize(
+    "ami_architecture, expected_default_instance_type, instance_info_err",
+    [
+        ("x86_64", "t2.xlarge", None),
+        ("arm64", "m6g.xlarge", None),
+        ("arm64", "m6g.xlarge", None),
+        ("arm64", "m6g.xlarge", "instance types do not exist"),
```

Comment:
For this failing tests I'd use another "wrong" instance type, instead of the working one "m6g.xlarge"

Reply:
This default instance type returned by the function will always be valid. The test case is to make sure the expected exception message generated when the default is unavailable in the given region. I think it makes sense to leave the test as is.